src/commands/actor/calculate-memory.ts

```diff
@@ -0,0 +1,111 @@
+import { join } from 'node:path';
+import process from 'node:process';
+
+import { calculateRunDynamicMemory } from '@apify/actor-memory-expression';
+
+import { ApifyCommand } from '../../lib/command-framework/apify-command.js';
+import { Flags } from '../../lib/command-framework/flags.js';
+import { CommandExitCodes } from '../../lib/consts.js';
+import { useActorConfig } from '../../lib/hooks/useActorConfig.js';
+import { error, info, success } from '../../lib/outputs.js';
+import { getJsonFileContent } from '../../lib/utils.js';
+
+export class ActorCalculateMemoryCommand extends ApifyCommand<typeof ActorCalculateMemoryCommand> {
```

Comment:
What is this magic? Why is it basically _extending itself_? 😅

Reply:
It's not extending itself but rather referencing itself in the extension class -> allows us to infer the types of the flags and args for `this.args`/`this.flags`